core/ui/EditorToolbar/file-import.tid

```diff
@@ -1,35 +1,46 @@
 title: $:/core/ui/EditorToolbar/file-import
 tags: $:/tags/EditorTools
-condition: [<targetTiddler>!has[type]] [<targetTiddler>type[text/vnd.tiddlywiki]]
+condition: [<targetTiddler>filter{$:/config/Editor/EnableImportFilter}]
 
-\define lingo-base() $:/language/Import/
+\procedure lingo-base() $:/language/Import/
 
-\define closePopupActions()
+\procedure closePopupActions()
 <$action-deletetiddler $filter="[title<importState>] [title<importTitle>]"/>
-\end
+\end closePopupActions
 
-\define replacement-text-image() [img[$title$]]
+\procedure tw5-ImageTemplate() [img[$(currentTiddler)$]]
+\procedure tw5-FileTemplate() [[$(currentTiddler)$]]
 
-\define replacement-text-file() [[$title$]]
+\procedure escape-regexp() [()<>\\]
+\function escape.title() [search-replace:g:regexp<escape-regexp>,[\$&]]
+\procedure markdown-ImageTemplate() ![](<#${ [<currentTiddler>escape.title[]] }$>)
+\procedure markdown-FileTemplate() [](<#${ [<currentTiddler>escape.title[]] }$>)
 
-\define postImportActions()
+\function is.markdown.tiddler() [all[]type[text/x-markdown]] [all[]type[text/markdown]]
+\function is.image() [get[type]prefix[image]]
+\function get.markdown.link() [is.image[]then<markdown-ImageTemplate>else<markdown-FileTemplate>]
+\function get.tw5.link() [is.image[]then<tw5-ImageTemplate>else<tw5-FileTemplate>] 
+\function get.link.template() [<storyTiddler>is.markdown.tiddler[]then<get.markdown.link>else<get.tw5.link>]
```

Comment:
With other global functions I did start to use a `tf.` prefix for `tiddlywiki function` similar to `tc-` classes. @Jermolene did merge several PRs, which did implement this pattern. It nicely avoids naming conflicts for user defined function names and it allows us to use `tf.*` functions in all possible usecases in filters. 

@Jermolene -- IMO there has not been a definitive decision / comment from your side yet.

Reply:
I agree that a formal consensus on naming global functions in the core would be helpful. However, in this case these are not global functions.

Reply:
see: https://tiddlywiki.com/#%24%3A%2Fcore%2Fmacros%2Ftag-picker

Reply:
@pmario what is the relevance of looking at that tiddler in this scenario?

Reply:
The link is **just an example**. I'll start a new thread under GH "discussion"